pkg/validation/internal/removed_apis.go

```diff
@@ -118,33 +115,84 @@ func validateDeprecatedAPIS(bundle *manifests.Bundle, versionProvided string) (e
 		}
 	}
 
-	// if the k8s value was informed and it is >=1.16 we should check
-	// if the k8s value was not informed we also should check since the
-	// check should occurs with any minKubeVersion value:
-	// - if minKubeVersion empty then means that the project can be installed in any version
-	// - if minKubeVersion any version defined it means that we are considering install
-	// in any upper version from that where the check is always applied
-	if !isVersionProvided || semVerVersionProvided.GE(semVerk8sVerV1betav1Deprecated) {
-		deprecatedAPIs := getRemovedAPIsOn1_22From(bundle)
-		if len(deprecatedAPIs) > 0 {
-			deprecatedAPIsMessage := generateMessageWithDeprecatedAPIs(deprecatedAPIs)
-			// isUnsupported is true only if the key/value OR minKubeVersion were informed and are >= 1.22
-			isUnsupported := semVerVersionProvided.GE(semVerK8sVerV1betav1Unsupported) ||
-				semverMinKube.GE(semVerK8sVerV1betav1Unsupported)
+	// Check if the bundle has removed APIs in 1.22
+	errs, warns = checkRemovedAPIS_1_22(bundle, semVerVersionProvided, semverMinKube, errs, warns)
+
+	// Check if the bundle has removed APIs in 1.25
+	errs, warns = checkRemovedAPIS_1_25(bundle, semVerVersionProvided, semverMinKube, errs, warns)
+
+	// Check if the bundle has removed APIs in 1.26
+	errs, warns = checkRemovedAPIS_1_26(bundle, semVerVersionProvided, semverMinKube, errs, warns)
+
+	return errs, warns
+}
+
+func checkRemovedAPIS_1_22(bundle *manifests.Bundle, semVerVersionProvided semver.Version, semverMinKube semver.Version, errs []error, warns []error) ([]error, []error) {
```

Comment:
do you think it could make sense to collapse `checkRemovedAPIS_1_22`, `checkRemovedAPIS_1_25`, `checkRemovedAPIS_1_26` into a single function?

Maybe you could do something like adding a routing function that takes in the version and gives you the related `getRemovedAPIs` (or bails). Then just add the version as a parameter to a general `checkRemovedAPIs(version, bundle, ...)`.

Just an idea. Not blocking atm. It's still fine with 3. But once we have more versions, this might stop scaling...